contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/GobblinDistcpJob.java

```diff
@@ -0,0 +1,217 @@
+            }
+        }
+    }
+
+    private void deleteDirOnExit(FileSystem fs, String target) throws Exception {
+        Path path = new Path(extractPathFromUrl(target));
+
+        fs.deleteOnExit(path);  // Delete the directory even if an exception occurs
+        info(path + " is scheduled to be deleted on exit.");
+    }
+
+    private void addPermissionsToParents(FileSystem fs, String target) throws Exception {
+        String pathFromRoot = removeLeadingSlash(extractPathFromUrl(target));
+        String parent = "";
+
+        for (String seg: pathFromRoot.split("/")) {
+            parent = parent + "/" + seg;
+            Path parentPath = new Path(parent);
+            FsPermission perm = fs.getFileStatus(parentPath).getPermission();
+            FsAction u = perm.getUserAction();
+            FsAction g = perm.getGroupAction();
+            FsAction o = perm.getOtherAction();
+            boolean changed = false;
+
+            if (props.getBoolean(VoldemortBuildAndPushJob.PUSH_CDN_READ_BY_GROUP, false)) {
+                if (!g.implies(FsAction.READ_EXECUTE)) {
+                    g = g.or(FsAction.READ_EXECUTE);
+                    changed = true;
+                }
+            }
+
+            if (props.getBoolean(VoldemortBuildAndPushJob.PUSH_CDN_READ_BY_OTHER, false)) {
+                if (!o.implies(FsAction.READ_EXECUTE)) {
+                    o = o.or(FsAction.READ_EXECUTE);
+                    changed = true;
+                }
+            }
+
+            if (changed) {
+                FsPermission desiredPerm = new FsPermission(u, g, o);
+                fs.setPermission(parentPath, desiredPerm);
+                assert fs.getFileStatus(parentPath).getPermission().equals(desiredPerm);
+                info("for path " + parent + ", permissions changed from: " + perm + " to: " + desiredPerm);
+            }
+        }
+    }
+
+    private String pickCDN() throws Exception {
+        List<String> pushClusters = props.getList(VoldemortBuildAndPushJob.PUSH_CLUSTER);
+        List<String> cdnClusters = props.getList(VoldemortBuildAndPushJob.PUSH_CDN_CLUSTER);
+
+        if (pushClusters.size() != cdnClusters.size()) {
+            warn("Cluster sizes are different! Will bypass CDN for push cluster " + destination);
+            return "";
+        }
+
+        int index = pushClusters.indexOf(destination);
+        assert index != -1;
+        String cdnCluster = cdnClusters.get(index);
+        if (cdnCluster.equals("null")) {
+            info("Will bypass CDN for push cluster " + destination);
+            return "";
+        }
+
+        if (!cdnCluster.matches(".*hdfs://.+:[0-9]{1,5}/?")) {
+            warn("Invalid URL format! Will bypass CDN for push cluster " + destination);
+            return "";
+        }
+        return removeTrailingSlash(cdnCluster);
+    }
+
+    private boolean prereqSatisfied(String cdnURL) throws Exception {
+        for (String namenode: props.getList("other_namenodes")) {
```

Comment:
Can you define this property name in 'VoldemortBuildAndPushJob' together with other cdn related properties?
It seem this one is very important to execute distcp job.